src/robot/libraries/BuiltIn.py

```diff
@@ -745,87 +761,154 @@ def should_be_equal_as_numbers(self, first, second, msg=None, values=True,
+            prv_str1 = str1.lower()
+            prv_str2 = str2.lower()
+        if not prv_str1.startswith(prv_str2):
             raise AssertionError(self._get_string_msg(str1, str2, msg, values,
                                                       'does not start with'))
 
-    def should_not_end_with(self, str1, str2, msg=None, values=True):
+    def should_not_end_with(self, str1, str2, msg=None, values=True, ignore_case=False):
         """Fails if the string ``str1`` ends with the string ``str2``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``str1`` and ``str2`` should be compared case-insensitively,
+        provided that ``str1`` and ``str2`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
+        if is_truthy(ignore_case) and is_string(str1) and is_string(str2):
+            str1 = str1.lower()
+            str2 = str2.lower()
         if str1.endswith(str2):
             raise AssertionError(self._get_string_msg(str1, str2, msg, values,
                                                       'ends with'))
 
-    def should_end_with(self, str1, str2, msg=None, values=True):
+    def should_end_with(self, str1, str2, msg=None, values=True, ignore_case=False):
         """Fails if the string ``str1`` does not end with the string ``str2``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``str1`` and ``str2`` should be compared case-insensitively,
+        provided that ``str1`` and ``str2`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
+        if is_truthy(ignore_case) and is_string(str1) and is_string(str2):
+            str1 = str1.lower()
+            str2 = str2.lower()
         if not str1.endswith(str2):
             raise AssertionError(self._get_string_msg(str1, str2, msg, values,
                                                       'does not end with'))
 
-    def should_not_contain(self, container, item, msg=None, values=True):
+    def should_not_contain(self, container, item, msg=None, values=True, ignore_case=False):
         """Fails if ``container`` contains ``item`` one or more times.
 
         Works with strings, lists, and anything that supports Python's ``in``
         operator. See `Should Be Equal` for an explanation on how to override
         the default error message with ``msg`` and ``values``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``container`` and ``item`` should be compared case-insensitively,
+        provided that ``container`` and ``item`` are string types.  See `Boolean
```

Comment:
If a container is list or list-like, shouldn't we actually convert items in it into lower case? More about that in a separate comment below, but most likely these docs need to be updated too.